topdown/http_test.go

```diff
@@ -2560,6 +2561,50 @@ func TestCertSelectionLogic(t *testing.T) {
 	}
 }
 
+func TestHTTPSendConcurrentRequests(t *testing.T) {
+
+	// run test server
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+
+	defer ts.Close()
+
+	t.Run("concurrent requests", func(t *testing.T) {
+		// add an inter-query cache
+		config, _ := iCache.ParseCachingConfig(nil)
+		interQueryCache := iCache.NewInterQueryCache(config)
+
+		m := metrics.New()
+		q := NewQuery(ast.MustParseBody(fmt.Sprintf(`http.send({"method": "get", "url": %q, "force_cache": true, "force_cache_duration_seconds": 10})`, ts.URL))).
+			WithInterQueryBuiltinCache(interQueryCache).
+			WithMetrics(m)
+
+		// Execute query concurrently and verify http.send inter-query cache hit metric is incremented.
+		wg := sync.WaitGroup{}
+
+		for i := 0; i < 5; i++ {
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+
+				_, err := q.Run(context.Background())
+				if err != nil {
+					t.Error(err)
+					return
+				}
+			}()
+		}
+
+		wg.Wait()
+
```

Comment:
Figuring out how to resolve https://github.com/open-policy-agent/opa/actions/runs/3432729846/jobs/5722316001

Reply:
Updated test.